### PR TITLE
DAOS-6880 Test: container/attribute.py test failed due to Test reports unsupported test status.

### PR DESCRIPTION
**Pull request by omar-ocampo**
Use DaosCommand for list attribute instead of pydaos.

Test-tag: container_attribute

Signed-off-by: Omar Ocampo <omar.ocampo.coronado@intel.com>

### CODE REVIEW

**Comment by dinghwah** [comment]
@Omar, for now, we should focus on the issue on daos_api.py issue. In order to catch coming weekend 1.2 CI, I push a workaround for attribute.py with my other JIAR last night. it came out good enough for the 37 attribute testcases.

**Comment by omar-ocampo** [comment]
Container attributes tests passing: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-5175/8/tests

File: src/tests/ftest/container/attribute.py
Comment (cibervicho) [inline src/tests/ftest/container/attribute.py:125]: Probably you should remove the tag 'container_attr' since you are defining a new common one (container_attribute) below.

Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:125]: Is there a way to search if a tag is being used by some process/automation? I'd like to avoid deleting a tag that is being used.

Comment (cibervicho) [reply src/tests/ftest/container/attribute.py:125]: Hmmm, probably locally with `./launch.py --list container_attr`

Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:125]: Alright, let me give it a look

Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:125]: Running avocado list --paginator=off --filter-by-tags=container_attr ./
Detected tests:
./container/attribute.py

Seems it is safe to implement your suggestion (y)

File: src/tests/ftest/container/attribute.py
Comment (berserk-fury) [inline src/tests/ftest/container/attribute.py:139]: is there a ticket to re-enable this?

Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:139]: Yes: https://jira.hpdd.intel.com/browse/DAOS-7093

Comment (berserk-fury) [reply src/tests/ftest/container/attribute.py:139]: Well, DAOS-7093 doesn't say that we would need to uncomment line L139 :)

Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:139]: Ticked was updated adding the information about container.list_attr() line.

**Comment by sylviachanoiyee** [comment]
PR looks good. Ready for merging.

**Comment by sylviachanoiyee** [comment]
@okulkarni , can you get a couple of people to review this?

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:62]: We follow the comments as shown below:
            indata (dict): Dict used to set attr
            outdata (dict): Dict obtained from list attr
            mode (str): sync or async, Defaults to sync.

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:71]: Maybe some comments on attribute structure can be explained here.

**Review by saurabhtandan — APPROVED** [review]
LGTM

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:113]: What is compared here? Comments please.

**Review by mjean308 — APPROVED** [review]

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:188]: Do we need these comments to be there? Looks like we are moving to use daos_cmd.

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:139]: Do we need this comment?

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:265]: Why these comments still in code?

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [inline src/tests/ftest/container/attribute.py:256]: Why these comments still in code when we move to daos_cmd?

**Comment by rpadma2** [comment]
Looks like this PR already has two approvals... In case the PR needs to be rebased, you can perform the changes. Otherwise, you can do the changes in a separate PR.

File: src/tests/ftest/container/attribute.py
Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:62]: Could you elaborate? Do you mean adding "Default to sync" within the docstrings?

File: src/tests/ftest/container/attribute.py
Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:71]: Sure, will add

File: src/tests/ftest/container/attribute.py
Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:188]: Yes, the intention is to keep the validation complete until pydaos bug (7093) is fixed. Another possibility is pydaos to not be used anymore in this test so we move completely to daos_cmd.

File: src/tests/ftest/container/attribute.py
Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:265]: When 7085 is resolved, this lines can be re-added

File: src/tests/ftest/container/attribute.py
Comment (omar-ocampo) [reply src/tests/ftest/container/attribute.py:256]: When 7093 is resolved, this lines can be re-added

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [reply src/tests/ftest/container/attribute.py:265]: Aren't we supposed to use skipForTicket pragma for this issue?

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [reply src/tests/ftest/container/attribute.py:256]: Same as above...

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [reply src/tests/ftest/container/attribute.py:188]: skipForTicket pragma should be used.

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [reply src/tests/ftest/container/attribute.py:62]: indata (dict):
outdata (dict): 
mode (str): 

Variable types should be added on the docstring and also if there is some defaults... Add them,

File: src/tests/ftest/container/attribute.py
Comment (rpadma2) [reply src/tests/ftest/container/attribute.py:256]: daos-gatekeeper may ask to add it before landing the PR. Please check with Sylvia.

**Comment by omar-ocampo** [comment]
@sylviachanoiyee , lets attend Ravi's comments in another review.

**Comment by sylviachanoiyee** [comment]
@okulkarni , I will land this PR but I want a follow up PR with :

1. All the comments regarding DAOS-7093 removed. 
2. Add a small basic container test that use pydaos container attr list (not daos_cmd) and use skipForTicket(DAOS-7093)
3. Add instructions in ticket DAOS-7093 where/how to re-enable the test

Thanks.

**Pull request merged**
